### PR TITLE
replace system.windows.interactivity with custom event to command attached property

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

Here is the last approach I wanted to prototype and get feedback on for removing `system.windows.interactivity`. This essentially replicates a tiny subset of the functionality in `system.windows.interactivity` to redirect events to commands - because sometimes we need to bind multiple events and commands to each uielement I modified an example from this article:
http://www.mobilemotion.eu/?p=1822

to handle comma separated lists event and command names.

There will be some perf penalty because of the string splitting and index searching but I think it will be quite minimal.

Perf and memory need to be profiled vs the existing solution. IE what happens when hundreds or thousands of nodes are drawn, what about dispose, are the events detached correctly? etc.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml [hunk 1]
```diff
@@ -88,19 +87,13 @@
                 <Rectangle
                     Fill="Transparent"
                     SnapsToDevicePixels="True"
-                    IsHitTestVisible="{Binding IsHitTestVisible}">
+                    IsHitTestVisible="{Binding IsHitTestVisible}"
+                        views:EventToCommand.Events="MouseEnter,MouseLeave,MouseLeftButtonDown"
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/UI/Themes/Modern/Ports.xaml]: note that now we pass a list of event names and command names  - we can do some validation that these need to have the same number of entries at runtime if we want.

File: src/DynamoCoreWpf/UI/HandlingEventTrigger.cs [hunk 2]
```diff
@@ -13,4 +19,113 @@ protected override void OnEvent(System.EventArgs eventArgs)
...
+                    var methodInfo = typeof(EventToCommand).GetMethod("OnEvent", BindingFlags.Static | BindingFlags.NonPublic);
+                    //TODO clear previous subscriptions?
+                    eventInfo.GetAddMethod().Invoke(obj,
+                        new object[] { Delegate.CreateDelegate(eventInfo.EventHandlerType, methodInfo) });
+                }
+            }
+        }
+
+        //TODO handle the events.
+        private static void OnEvent(object sender, RoutedEventArgs e)
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/UI/HandlingEventTrigger.cs]: this is called when the event is called.

File: src/DynamoCoreWpf/UI/HandlingEventTrigger.cs [hunk 3]
```diff
@@ -13,4 +19,113 @@ protected override void OnEvent(System.EventArgs eventArgs)
...
+
+        public static void SetEvents(DependencyObject obj, string value)
+        {
+            //parse string "eventName1,eventName2";
+
+            var eventNames = parseEventNames(value);
+            obj.SetValue(EventProperty, eventNames);
+        }
+
+        public static void EventPropertyChangedCallback(DependencyObject obj, DependencyPropertyChangedEventArgs args)
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/UI/HandlingEventTrigger.cs]: this is called when the event property is set in the xaml.

File: src/DynamoCoreWpf/UI/HandlingEventTrigger.cs [hunk 4]
```diff
@@ -13,4 +19,113 @@ protected override void OnEvent(System.EventArgs eventArgs)
...
+
+    //forked from: http://www.mobilemotion.eu/?p=1822
+
+
+    public class EventToCommand
+    {
+        public static readonly DependencyProperty CommandProperty =
+            DependencyProperty.RegisterAttached("Commands", typeof(object), typeof(EventToCommand));
+
+        public static ICommand[] GetCommands(DependencyObject obj)
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/UI/HandlingEventTrigger.cs]: these setters and accessors are not used when set from xaml - so unless someone else sets these dep properties from code behind it's unlikely to be used - but it's required for making an attached dep property.

**Comment by aparajit-pratap** [comment]
Out of the 3 potential solutions I find this one to be the simplest and cleanest one however, I'm not sure about the pros and cons of the first solution as compared to this one.

**Pull request closed without merging**